lib/elixir/src/elixir_module.erl

```diff
@@ -229,6 +219,28 @@ build(Line, File, Module, Lexical) ->
 
   {Data, Defs, Ref}.
 
+compile_definition_attributes(#{module := Module} = Env, Kind, Name, Args, _Guards, _Body) ->
```

Comment:
We can move this logic to the `Module` module.

Reply:
So in a nutshell, we will have `Module.compile_definition_attributes` and `:elixir_module.delete_definition_attributes`. We will specify which one to call based on the bootstrap value when the module is defined ([the previous OnDefinition field here](https://github.com/elixir-lang/elixir/pull/7113/files#diff-b11934843d48f08ba2bdb4d4965b812bL204)).